modules/face/samples/sampleKazemi.cpp

```diff
@@ -0,0 +1,124 @@
+#include "opencv2/imgcodecs.hpp"
+
+using namespace std;
+using namespace cv;
+
+int main(int argc,char** argv){
+    //Give the path to the directory containing all the files containing data
+    if(argc!=2){
+        String error_message = "No directory specified to give names of the files to be read\n Aborting....";
+        cerr<<error_message;
+        return 0;
+    }
+
+    String directory = argv[1];
+    //create a vector to store names of files in which annotations
+    // and image names are found
+    /*The format of the file containing annotations should be of following format
+        /data/abc
+        123.45,345.65
+        321.67,543.89
+
+        The above format is similar to HELEN dataset which is used for training model
+     */
+
+    vector<String> filenames;
+    //reading the files from the given directory
+    glob(directory,filenames);
+    //create a vector to store image names
+    vector<String> imagenames;
+    //create object to get landmarks
+    vector< vector<Point2f> > trainlandmarks;
+    //create a pointer to call the base class
+    Ptr<cv::face::FacemarkKazemi> facemark=cv::face::create();
+    //gets landmarks and corresponding image names in both the vectors
+    facemark->getData(filenames,trainlandmarks,imagenames);
+    //vector to store images
+    vector<Mat> trainimages;
+    for(unsigned long i=0;i<imagenames.size();i++){
+        string imgname = imagenames[i].substr(0, imagenames[i].size()-1);
+        string img =string(imgname);
+        img.append(".jpg");
+        Mat src = imread(img);
+        if(!src.data){
+            string error = "Image"+img+"not found\n.Aborting...";
+            cerr<<error<<endl;
+            return 0;
+        }
+        trainimages.push_back(src);
+    }
+    //Now you can scale your images your landmarks and your images
+    //to a lower size as training on large images would induce an overhead.
+    //Let size taken be (460,460) which can be taken as per your choice
+    Size s =Size(460,460);
+    //Now scale data according to the size selected
+    facemark->scaleData(trainlandmarks,trainimages,s);
+    //calculate mean shape
+    facemark->calcMeanShape(trainlandmarks,trainimages);
+    /*Now train data using training function which is yet to be built*/
+
+
+
+    /*For detection follow the following procedure*/
+
+    //Load the image in which you want to detect landmarks
+    Mat src = imread("100040721_1.jpg");
+    if(!src.data){
+        string error = "Image not found\n.Aborting...\n";
+        cerr<<error<<endl;
+        return 0;
+    }
+    //you can resize your image according to your requirements. Though it is not necessary here.
+    resize(src,src,Size(300,300));//optional
+    //vector to store faces in the image
+    vector<Rect> faces;
+    //vector to store facelandmarks
+    vector< vector<Point2f> > facesLandmarks;
+    //get bounding rectangle for all faces in the image
+    enum face_cascade_classifier{ HAAR = 0, LBP = 1 };
```

Comment:
This enum should go on top globally. Enums inside the code always seems weird to me! I would also add the enum in the class, not in the sample. Also make enum meaningful like

enum face_cascade_classifier{ FACE_HAAR = 0, FACE_LBP = 1 };